src/ert/storage/local_ensemble.py

```diff
@@ -180,9 +186,13 @@ def realizations_initialized(self, realizations: List[int]) -> bool:
 
         return all((responses[real] or parameters[real]) for real in realizations)
 
-    def get_realization_list_with_responses(self) -> List[int]:
+    def get_realization_list_with_responses(
+        self, key: Optional[str] = None
+    ) -> List[int]:
         return [
-            idx for idx, b in enumerate(self.get_realization_mask_with_responses()) if b
+            idx
+            for idx, b in enumerate(self.get_realization_mask_with_responses(key))
+            if b
         ]
```

Comment:
Since `self.get_realization_mask_with_responses(key)` returns a boolean numpy array, I think it may be a good idea to rewrite this as:
```python
    mask = self.get_realization_mask_with_responses(key)
    return np.where(mask)[0].tolist()
```